### DISCUSSION TITLE
Copy local folder to broker pod in Stimizi

### DISCUSSION THREAD

**Discussion by barryzhounb** [strimzi/strimzi-kafka-operator#4525] [General]
In order to run some kafka cli commands inside broker pod, so I want to copy my configuration files to pod.

This is all pods in my OCP
```
oc get pods

NAME                                             READY     STATUS    RESTARTS   AGE
dlake-cluster-entity-operator-7b5f98876d-h9pc5   3/3       Running   0          15m
dlake-cluster-kafka-0                            2/2       Running   0          16m
dlake-cluster-kafka-1                            2/2       Running   0          16m
dlake-cluster-kafka-2                            2/2       Running   0          16m
dlake-cluster-zookeeper-0                        1/1       Running   0          16m
dlake-cluster-zookeeper-1                        1/1       Running   0          16m
dlake-cluster-zookeeper-2                        1/1       Running   0          16m
```

Now I try to copy folder client-ca at local to broker pod dlake-cluster-kafka-0. But I get some error that comnplains permission issue.

```
oc rsync client-ca dlake-cluster-kafka-0:/opt/kafka -c kafka

building file list ... done
client-ca/

sent 180 bytes  received 26 bytes  137.33 bytes/sec
total size is 10633  speedup is 51.62
rsync: recv_generator: mkdir "/opt/kafka/client-ca" failed: Permission denied (13)
*** Skipping any contents from this failed directory ***
rsync error: some files could not be transferred (code 23) at /AppleInternal/BuildRoot/Library/Caches/com.apple.xbs/Sources/rsync/rsync-54.120.1/rsync/main.c(996) [sender=2.6.9]
error: exit status 23
```

My question is, how can I fix that permission issue? Or is there any ways to copy folder or files to broker pod?

Thanks

**Comment by scholzj** [comment]
The container runs under a user with minimal privileges. So you cannot copy files anywhere. You can write only to `/tmp` and to the volume which stores the Kafka logs. You should also ideally avoid copying anything into the container or running applications from it. Just run your own container to run the tools there, e.g.:

`kubectl run my-tools -ti --image=quay.io/strimzi/kafka:0.21.1-kafka-2.7.0 --rm=true --restart=Never -- bash`
